src/pages/home/report/ReportActionCompose/ComposerWithSuggestions.js

```diff
@@ -0,0 +1,668 @@
+                const parentReportAction = lodashGet(parentReportActions, [parentReportActionID], {});
+                const lastReportAction = _.find([...reportActions, parentReportAction], (action) => ReportUtils.canEditReportAction(action));
+
+                if (lastReportAction !== -1 && lastReportAction) {
+                    Report.saveReportActionDraft(reportID, lastReportAction.reportActionID, _.last(lastReportAction.message).html);
+                }
+            }
+        },
+        [isKeyboardShown, isSmallScreenWidth, parentReportActions, report, reportActions, reportID, submitForm, suggestionsRef, valueRef],
+    );
+
+    const onSelectionChange = useCallback(
+        (e) => {
+            LayoutAnimation.configureNext(LayoutAnimation.create(50, LayoutAnimation.Types.easeInEaseOut, LayoutAnimation.Properties.opacity));
+
+            if (suggestionsRef.current.onSelectionChange(e)) {
+                return;
+            }
+
+            setSelection(e.nativeEvent.selection);
+        },
+        [suggestionsRef],
+    );
+
+    const updateShouldShowSuggestionMenuToFalse = useCallback(() => {
+        if (!suggestionsRef.current) {
+            return;
+        }
+        suggestionsRef.current.updateShouldShowSuggestionMenuToFalse(false);
+    }, [suggestionsRef]);
+
+    const setShouldBlockSuggestionCalc = useCallback(() => {
+        if (!suggestionsRef.current) {
+            return false;
+        }
+        return suggestionsRef.current.setShouldBlockSuggestionCalc(true);
+    }, [suggestionsRef]);
+
+    /**
+     * Focus the composer text input
+     * @param {Boolean} [shouldDelay=false] Impose delay before focusing the composer
+     * @memberof ReportActionCompose
+     */
+    const focus = useCallback((shouldDelay = false) => {
+        // There could be other animations running while we trigger manual focus.
+        // This prevents focus from making those animations janky.
+        InteractionManager.runAfterInteractions(() => {
+            if (!textInputRef.current) {
+                return;
+            }
+
+            if (!shouldDelay) {
+                textInputRef.current.focus();
+            } else {
+                // Keyboard is not opened after Emoji Picker is closed
+                // SetTimeout is used as a workaround
+                // https://github.com/react-native-modal/react-native-modal/issues/114
+                // We carefully choose a delay. 100ms is found enough for keyboard to open.
+                setTimeout(() => textInputRef.current.focus(), 100);
+            }
+        });
+    }, []);
+
+    const setUpComposeFocusManager = useCallback(() => {
+        // This callback is used in the contextMenuActions to manage giving focus back to the compose input.
+        // TODO: we should clean up this convoluted code and instead move focus management to something like ReportFooter.js or another higher up component
```

Comment:
Could you defined exactly the problem and what we need to do so we can create an issue for it and remove this TODO?

Reply:
same here, I think this is good now